school-library/persons.rb

```diff
@@ -31,4 +33,9 @@ def can_use_services?
 
     false
   end
+
+  def add_rental(rental)
+    @rental.push(rental)
+    rental.person = self
+  end
```

Comment:
@DeliceLydia, 
I believe I added the method ffor the add_rentals here as well as in book.rb file

Reply:
You indeed added the add rental method to both the classes as I suggested, but it's not fulfilling its purpose correctly. It takes an existing rental. 

Instead, we want to add a new rental. So it should take a book in this case (because we're on the person class) and a date, and from there it's just a matter of creating the rental with the Rental class. On the initialize method of the rental, the rental will add itself to both the person and the book.

```
def add_rental(book, date)
  Rental.new(book, date, self)
end
```